searx/engines/base.py

```diff
@@ -0,0 +1,121 @@
+#!/usr/bin/env python
+
+"""
+ BASE (Scholar publications)
+
+ @website     https://base-search.net
+ @provide-api yes with authorization (https://api.base-search.net/)
+
+ @using-api   yes
+ @results     XML
+ @stable      ?
+ @parse       url, title, publishedDate, content
+ More info on api: http://base-search.net/about/download/base_interface.pdf
+"""
+
+from lxml import etree
+from urllib import urlencode
+from searx.utils import searx_useragent
+from cgi import escape
+from datetime import datetime
+import re
+
+categories = ['science']
+
+base_url = 'https://api.base-search.net/cgi-bin/BaseHttpSearchInterface.fcgi'\
+           + '?func=PerformSearch&{query}&boost=oa&hits={hits}&offset={offset}'
+
+# engine dependent config
+paging = True
+number_of_results = 10
+
+
+def request(query, params):
+    # shortcuts for advanced search
+    shorcut_dict = {
+        'format': 'dcformat',
+        'author': 'dccreator',
+        'collection': 'dccollection',
+        'hdate': 'dchdate',
+        'contributor': 'dccontributor',
+        'coverage': 'dccoverage',
+        'date': 'dcdate',
+        'abstract': 'dcdescription',
+        'urls': 'dcidentifier',
+        'language': 'dclanguage',
+        'publisher': 'dcpublisher',
+        'relation': 'dcrelation',
+        'rights': 'dcrights',
+        'source': 'dcsource',
+        'subject': 'dcsubject',
+        'title': 'dctitle',
+        'type': 'dcdctype'
+    }
+# replace shortcuts with API advanced search keywords
+    for key in shorcut_dict.keys():
+        query = re.sub(str(key), str(shorcut_dict[key]), query)
+
+# basic search
+    offset = (params['pageno'] - 1) * number_of_results
+
+    string_args = dict(query=urlencode({'query': query}),
+                       offset=offset,
+                       hits=number_of_results)
+
+    search_url = base_url
+
+    params['url'] = search_url.format(**string_args)
+
+    params['headers']['User-Agent'] = searx_useragent()
+    return params
+
+
+def response(resp):
+    results = []
+
+    search_results = etree.XML(resp.content)
+
+    for entry in search_results.xpath('./result/doc'):
+        content = "No description available"
+
+        date = datetime.now()  # needed in case no dcdate is available for an item
+        for item in entry:
+            if item.attrib["name"] == "dchdate":
+                harvestDate = item.text
+
+            if item.attrib["name"] == "dcdate":
```

Comment:
In this loop you are checking the value of only the `name` key in all of the `if`s. Using `elif` would be much better because this code would be skipped if one of the conditions were true. The same applies to the next `if` below this one. In the last two cases using 'elif` is the appropiate choice. :)